### PR TITLE
Automatically find cartridges and provide checkbox for selecting & masking of password field in options view.

### PR DESCRIPTION
**Pull request by ghgofort**
- Removed the atom-text-editor for entering cartridges and added checkboxes for any cartridges that are found in the project folders.  This is done by checking if the name matches the required naming scheme for a cartridge and checking to see if the cartridge sub-directory exists. If the cartridge is found in the saved current options, then it is selected for the user. This only searches the project directories, and their immediate children directories to keep performance in mind.
- Replaced the atom-text-editor for the password field with an HTML input field type=password in order to mask the password as you type it in.
- Set the options view to scroll to the error display div when an error is displayed so that it is immediately apparent what the issue is when a user is unable to save any updates to the options.

If the pull request for the 'save credentials' functionality is merged, I will pull it into my fork to fix merge issues, and then resubmit.

### CODE REVIEW

File: lib/OptionsView.js [hunk 1]
```diff
@@ -140,17 +148,127 @@ export default class OptionsView {
...
+                }
+            });
+        }
+    }
+    checkDirLevel(dir) {
+        dir.getEntries((err, children) => {
+            if (err) {
+                this.setError(err);
+            } else {
+                let parent = dir.getPath().split('/').pop();
```
Comment (SqrTT) [inline lib/OptionsView.js]: You would use `path.sep` instead of `'/'`

File: lib/OptionsView.js [hunk 2]
```diff
@@ -140,17 +148,127 @@ export default class OptionsView {
...
+        let dirs = atom.project.getDirectories();
+        if (dirs.length) {
+            dirs.forEach((dir) => {
+                let path = dir.getPath();
+                let name = path.split('/').pop();
+                if (this.isCartridge(dir) &&
+                !this.cartridges.all.includes(name)) {
+                    this.createCartridgeCheckbox(dir, '');
+                } else {
+                    let prs = this.checkDirLevel(dir);
```
Comment (SqrTT) [inline lib/OptionsView.js]: Does `prs` is used somewhere?

**Comment by SqrTT** [comment]
Hi @ghgofort, thanks for contributing.
There are several issues that should be addressed.
* Search for cartridges should be deep than 2 levels, for instance, in my projects cartridges is in `cartridges` directory 'cartridges/app_storefront/cartridge/..', etc. so it doesn't work for me.
* As I remember native inputs doesn’t work properly in Atom, so now this is fixed? Also you should not use harcoded CSS styles but use system instead. Because it looks weird with different themes. for instance:

* also please address conflicts

Thanks you.

**Comment by ghgofort** [comment]
@SqrTT 
- I have addressed all of the items that you mentioned except for the use of an input field.  This does work with some special handling for the Backspace character.  Please let me know if you see other things that need to be addressed in my pull, or if you have any questions about why I did something.

**Comment by SqrTT** [comment]
@ghgofort 

Quick test shows that upload doesn't work for structure like on picture and doesn't upload `modules` folder. Probably I could fix by myself, but I really have a lack of time.



Thanks.

**Comment by ghgofort** [comment]
@SqrTT - I have updated and tested with the folder structure you have listed.  I have not done anything about the modules folder yet since this isn't really a cartridge and I am thinking of a couple of different ways to deal with additional folders like this. I will commit to this pull request again once this is complete.

**Comment by ghgofort** [comment]
@SqrTT I will re-open this when I add in a separate list of folders for uploading.  Not a lot of time lately.

**Pull request closed without merging**